tests/robottelo/test_helpers.py

```diff
@@ -133,3 +141,317 @@ def test_multiple_dicts_converted_to_storage(self):
+                data = {
+                    "id": 123456,
+                    "status": status,
+                    "resolution": "",
+                    "target_milestone": "Unspecified",
+                    "flags": [{"status": "+", "name": "sat-6.6.0"}],
+                    "clones": []
+                }
+                self.assertFalse(is_open("BZ:123456", data))
+
+    def test_bz_is_open_using_dupe_data_higher_version(self):
+        """Assert that if BZ has a dupe, the dupe data is considered.
+        The dupe is CLOSED/ERRATA but on a future version, no clones for
+        backport the solution."""
+        data = {
+            "id": 123456,
+            "status": "CLOSED",
+            "resolution": "DUPLICATE",
+            "target_milestone": "Unspecified",
+            "flags": [],
+            "dupe_data": {
+                "id": 999999,
+                "status": "CLOSED",
+                "resolution": "ERRATA",
+                "target_milestone": "Unspecified",
+                "flags": [{"status": "+", "name": "sat-6.7.z"}]
+            }
+        }
+        self.assertTrue(is_open("BZ:123456", data))
+
+    def test_bz_is_not_open_using_dupe_data_lower_version(self):
+        """Assert that if BZ has a dupe, the dupe data is considered.
+        The dupe is CLOSED/ERRATA in a previous version."""
+        data = {
+            "id": 123456,
+            "status": "CLOSED",
+            "resolution": "DUPLICATE",
+            "target_milestone": "Unspecified",
+            "flags": [],
+            "dupe_data": {
+                "id": 999999,
+                "status": "CLOSED",
+                "resolution": "ERRATA",
+                "target_milestone": "Unspecified",
+                "flags": [{"status": "+", "name": "sat-6.3.z"}]
+            }
+        }
+        self.assertFalse(is_open("BZ:123456", data))
+
+    def test_bz_is_open_using_dupe_data_clones(self):
+        """Assert that if BZ has a dupe, the dupe data is considered.
+        The dupe is CLOSED/ERRATA but on a future version, and also has
+        a clone that is the backport for the current server version
+        and the clone backport is not yet applied."""
+        data = {
+            "id": 123456,
+            "status": "CLOSED",
+            "resolution": "DUPLICATE",
+            "target_milestone": "Unspecified",
+            "flags": [],
+            "dupe_data": {
+                "id": 999999,
+                "status": "CLOSED",
+                "resolution": "ERRATA",
+                "target_milestone": "Unspecified",
+                "flags": [{"status": "+", "name": "sat-7.7.z"}],
+                "clones": [
+                    {
+                        "id": 999999,
+                        "status": "POST",
+                        "resolution": "",
+                        "target_milestone": "Unspecified",
+                        "flags": [{"status": "+", "name": "sat-6.6.z"}],
+                        "clones": []
```

Comment:
Does bugzilla ensure `clones` to be symmetrical relationship?
In other words, is it guaranteed that if BZ 1 clones BZ 2, and BZ 2 clones BZ 3, then BZ 3 "clones" field will contain both BZ 1 and 2?

If not, then perhaps "clones" field handling should be done recursively or in the loop, to ensure all clones are found.

Reply:
BUgzilla returns only a list `clone_ids` holding the direct clones, if the clone has a clone than you are right and we need to follow it.

Reply:
We will not follow clones anymore, @mirekdlugosz this responsibility is being deferred to the nagger script.